llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1411,11 +1411,19 @@ bool VPlanTransforms::tryAddExplicitVectorLength(VPlan &Plan) {
   // The transform updates all users of inductions to work based on EVL, instead
   // of the VF directly. At the moment, widened inductions cannot be updated, so
   // bail out if the plan contains any.
-  if (any_of(Header->phis(), [](VPRecipeBase &Phi) {
-        return (isa<VPWidenIntOrFpInductionRecipe>(&Phi) ||
-                isa<VPWidenPointerInductionRecipe>(&Phi));
-      }))
+  bool IncludeWidenInduction = any_of(Header->phis(), [](VPRecipeBase &Phi) {
+    return (isa<VPWidenIntOrFpInductionRecipe>(&Phi) ||
+            isa<VPWidenPointerInductionRecipe>(&Phi));
+  });
+  // FIXME: Remove this once we can transform (select header_mask, true_value,
+  // false_value) into vp.merge.
+  bool IncludeOutloopReduction = any_of(Header->phis(), [&](VPRecipeBase &Phi) {
```

Comment:
`ContainsOutLoopReductions`? Move comment to the bail out?

Reply:
7d2a8c4f77c458389e56bb51c9af83595dea8862
Ditto.